### DISCUSSION TITLE
Customizing the default 'edgy' mode layout

### DISCUSSION THREAD

**Discussion by theesfeld** [LazyVim/LazyVim#4087] [Q&A] [answered]
I have found 'edgy' to be the exact functionality I am looking for. I can quickly edit a file from the command line with `nvim filename`, and if I want to use vim as a full-fledged IDE I simply <SPC>-u-e ...but id like to customize what plugins open in the sidebars and their sizes when edgy opens.

I know it's a basic n00b question, but im a basic n00b with lazyvim, and im not entirely sure what file to edit or create. the documentation is kind of lacking in explaining this customization.

any help will be appreciated.

thanks

**Comment by folke** [comment]
Just read the edgy.nvim docs?

**Comment by folke** [comment]
I don't write those docs for fun. I write them so that I don't get questions asked by hundreds of people every day...

**Comment by dotfrag** [comment, marked as answer]
You have to take a look at what is already configured, and then override it with your own plugin spec.

This is where edgy config is specified: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/extras/ui/edgy.lua

Take a look at the `opts` key..

To override the config you need to first look into `edgy.nvim` docs, to see what you can do, and then LazyVim docs [here](https://www.lazyvim.org/), more specifically the [example spec](https://www.lazyvim.org/configuration/examples) which shows how to override a plugin spec.

The override file has to go in your nvim config, under `lua/plugins` directory. For example: `lua/plugins/edgy.lua`.

**Reply by vdegenne** [reply]
Do you know if it is possible to focus the filesystem by default when edgy opens? when I hit `<leader>ue` all the windows open but the outline one on the right is the one to get the focus.

**Reply by dotfrag** [reply]
I don't use edgy, sorry. I can see there's an [API](https://github.com/folke/edgy.nvim?tab=readme-ov-file#-api) you might be able to hack around.

**Reply by vdegenne** [reply]
@dotfrag Thanks that's perfect, what I was looking for :-)
